### PR TITLE
Add docs for DynamoDB and Kinesis stream setup

### PR DESCRIPTION
**Pull request by pmuens**
Add "mock plugin directory" (where we can later on easily start the implementation in when we have enough feedback) and docs how to setup stream support for Kinesis and DynamoDB.

Furthermore links to the corresponding GitHub issues are added.

@mthenw can you take a look if the stream syntax is correct? Thanks!

Corresponding issues are:
[DynamoDB](https://github.com/serverless/serverless/issues/1441) and [Kinesis](https://github.com/serverless/serverless/issues/1608)

/cc @eahefnawy @flomotlik

### CODE REVIEW

**Comment by pmuens** [comment]
@mthenw good points! Thanks for the feedback! Will update it accordingly.

File: docs/guide/overview-of-event-sources.md [hunk 1]
```diff
@@ -140,3 +147,47 @@ functions:
...
+```yml
+# serverless.yml
+
+resources
+  Resources:
+    mapping:
+      Type: AWS::Lambda::EventSourceMapping
+      Properties:
+        BatchSize: 10
+        EventSourceArn: "arn:aws:kinesis:<region>:<aws-account-id>:stream/<stream-name>"
```
Comment (eahefnawy) [inline docs/guide/overview-of-event-sources.md]: I think most users would setup the stream using custom resources as well. In that case they wouldn't have the arn available and they'd use `Ref` in that case. We should at least have a comment or note that mentions that case when they're creating the stream via Serverless custom resources.

File: docs/guide/overview-of-event-sources.md [hunk 2]
```diff
@@ -140,3 +147,47 @@ functions:
...
+```yml
+# serverless.yml
+
+resources
+  Resources:
+    mapping:
+      Type: AWS::Lambda::EventSourceMapping
+      Properties:
+        BatchSize: 10
+        EventSourceArn: "arn:aws:dynamodb:<region>:<aws-account-id>:table/<table-name>/stream/<stream-name>"
```
Comment (eahefnawy) [inline docs/guide/overview-of-event-sources.md]: same comment as above, although I'm not sure you can create a dynamoDB stream as a CF resource. Hmmm. I'd just mention it for kinesis stream and leave this as it is.

**Comment by pmuens** [comment]
@eahefnawy @flomotlik done. Added a note that the user can add the stream beforehand and reference it via `Fn::GetAtt` (`Ref` won't return the `arn`, therefore `Fn::GetAtt`).

**Comment by eahefnawy** [comment]
@pmuens awesome G2M!

File: docs/guide/overview-of-event-sources.md (hunk 2 above)
Comment (pouledodue) [inline docs/guide/overview-of-event-sources.md]: there is some hack here : https://forums.aws.amazon.com/thread.jspa?messageID=680436&#680436

is it usable

**Pull request merged**